From a7f56634e47cdf8987be595b4ac024913e85da1d Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Mon, 18 Oct 2021 03:55:15 +0200
Subject: [PATCH 426/446] arm64: dts: rk3399-pinephone-pro: Fix goodix
 toucscreen interrupts

The interrupt type read from the screen does not work well. It generates
constant stream of interrupts.

Change to rising edge interrupt, and enforce it in the driver via
'use-dt-irq-flags'.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 420a86d0c..d07a06afa 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -594,7 +594,8 @@ touchscreen@14 {
 		compatible = "goodix,gt917s";
 		reg = <0x14>;
 		interrupt-parent = <&gpio3>;
-		interrupts = <RK_PB5 IRQ_TYPE_LEVEL_HIGH>;
+		interrupts = <RK_PB5 IRQ_TYPE_EDGE_RISING>;
+		use-dt-irq-flags;
 		irq-gpios = <&gpio3 RK_PB5 GPIO_ACTIVE_HIGH>;
 		reset-gpios = <&gpio3 RK_PB4 GPIO_ACTIVE_HIGH>;
 		AVDD28-supply = <&vcc3v0_touch>;
-- 
2.31.1

